modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSecondaryIndexTest.java

```diff
@@ -993,4 +993,60 @@ public void testSaturatedBoundsHashIndex() {
 
         assertQuery("SELECT * FROM t200 WHERE val = 1024").returnNothing().check();
     }
+
+    @Test
+    public void testScanBooleanField() {
+        try {
+            sql("CREATE TABLE t(i INTEGER PRIMARY KEY, b BOOLEAN)");
+            sql("INSERT INTO t VALUES (0, TRUE), (1, TRUE), (2, FALSE), (3, FALSE), (4, null)");
+            sql("CREATE INDEX t_idx ON t(b)");
+
+            assertQuery("SELECT i FROM t WHERE b = TRUE")
+                    .matches(containsIndexScan("PUBLIC", "T", "T_IDX"))
+                    .returns(0)
+                    .returns(1)
+                    .check();
+
+            assertQuery("SELECT i FROM t WHERE b = FALSE")
+                    .matches(containsIndexScan("PUBLIC", "T", "T_IDX"))
+                    .returns(2)
+                    .returns(3)
+                    .check();
+
+            assertQuery("SELECT i FROM t WHERE b IS TRUE")
+                    .matches(containsIndexScan("PUBLIC", "T", "T_IDX"))
+                    .returns(0)
+                    .returns(1)
+                    .check();
+
+            assertQuery("SELECT i FROM t WHERE b IS FALSE")
+                    .matches(containsIndexScan("PUBLIC", "T", "T_IDX"))
+                    .returns(2)
+                    .returns(3)
+                    .check();
+
+            // Support index scans for IS TRUE, IS FALSE but not for IS NOT TRUE, IS NOT FALSE,
+            // since it requires multi-bounds scan and may not be effective.
```

Comment:
@lowka We can rewrite `a IS NOT TRUE` to `a = FALSE or a is NULL`. If I understand correctly, then this will require several index scans, which (at least for boolean) are less efficient than a single table scan :thinking:
What is the bug here?

Reply:
You are correct there is no problem when `b` is not nullable. and WHERE b = true and WHERE b is not false be have the same way in that case.

Reply:
> several index scans, which (at least for boolean) seems less efficient than a single table scan

why do you think so?

Reply:
Filed a ticket [1] to improve this behavior in the future, tests are muted with a link to it.

[1] https://issues.apache.org/jira/browse/IGNITE-19964